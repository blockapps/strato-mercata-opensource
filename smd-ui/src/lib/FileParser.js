export function getImportStatements(source) {
    const splittedSource = source.split("\n")
    // TODO: This does not guard against multiline strings or comments
    const newImports = splittedSource.filter((line) => /^\s*import /.test(line))
    const newFileList = []
    newImports.map((item) => {
        const splittedItem = item.split(' ')
        const filename = splittedItem.filter((word) => word.includes('.sol'))
        newFileList.push(filename.length > 0 && filename[0].replace(/['";]+/g, ''))
        return item
    })
    return newFileList
}

export function replaceImportStatementsWithSource(source, importsCode, tabData) {
    const splittedSource = source.split("\n")
    const filteredList = splittedSource.map((item) => {
        const code = importsCode.filter(
            (codeItem) =>
                item.includes(codeItem.title)
        )
        return code.length > 0 ? getFileAndReplaceWithImport(code[0].text, tabData) : item }
    )
    filteredList.join("\r\n")
    return filteredList
}

export function getFileAndReplaceWithImport(code, tabData) {
    const newFileList = getImportStatements(code)
    if (newFileList.length === 0) {
        return code
    }
    const items = tabData.filter((item) =>
        newFileList.includes(item.title)
    )
    const replacedCode = replaceImportStatementsWithSource(code, items, tabData)
    return replacedCode.join("\r\n")
}
