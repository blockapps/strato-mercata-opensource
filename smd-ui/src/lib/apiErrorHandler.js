export function handleApiError(response) {
  return new Promise(function(resolve,reject){
    if(response.error) {
      reject(response.error);
    }
    else {
      resolve(response);
    }
  })
}
