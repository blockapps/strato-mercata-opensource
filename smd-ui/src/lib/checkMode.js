import { env } from "../env";

export function isOauthEnabled() {
  return env.OAUTH_ENABLED
}