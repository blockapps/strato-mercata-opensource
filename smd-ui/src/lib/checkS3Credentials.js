import { env } from "../env";

export function isS3Available() {
  return env.EXT_STORAGE_ENABLED === 'true'
}
