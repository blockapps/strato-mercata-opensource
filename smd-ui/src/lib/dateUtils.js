import moment from 'moment';

export function parseDateFromString(timeAsString) {
  const utcDate = timeAsString.replace(' UTC', 'Z')
  const formattedDate = moment(utcDate).format('YYYY-MM-DD hh:mm:ss A')
  return formattedDate
}

export function parseDateFromTimestamp(timestamp) {
  return parseDateFromString(moment.unix(timestamp).utc().toString());
}
