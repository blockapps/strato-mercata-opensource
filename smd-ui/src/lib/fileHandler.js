export function downloadFile(name, content) {
    var element = document.createElement('a');
    element.setAttribute('href', 'data:text/plain;charset=utf-8,' + encodeURIComponent(content));
    element.setAttribute('download', name);
    element.style.display = 'none';
    document.body.appendChild(element);
    element.click();
    document.body.removeChild(element);
}

export function downloadPDFFile(name, content) {
    var element = document.createElement('a');
    element.setAttribute('href', content);
    element.setAttribute('download', name);
    element.style.display = 'none';
    document.body.appendChild(element);
    element.click();
    document.body.removeChild(element);
}