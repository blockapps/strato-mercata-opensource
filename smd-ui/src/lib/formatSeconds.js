import React from "react";
export function sec2Date(sec){
    if (sec === undefined || sec <= 0 ){
        return <div></div>;
    }
    let seconds = sec;
    const days = Math.floor(seconds / (3600*24));
    const daysString = days.toString().padStart(2, "0");
    seconds  -= days*3600*24;
    const hrs = Math.floor(seconds / 3600);
    const hoursString = hrs.toString().padStart(2, "0");
    seconds  -= hrs*3600;
    const mnts = Math.floor(seconds / 60);
    const minutesString = mnts.toString().padStart(2, "0");
    seconds  -= mnts*60;
    const secondsString = Math.floor(seconds).toString().padStart(2, "0");
    const ret = <div>
        Uptime {days > 0 ? <span>{daysString}d:</span> : null}
         <span>
             {hoursString}:
        </span>
        <span>
             {minutesString}:

        </span>
        <span>
             {secondsString}
        </span>
    </div>
    return ret;
}
