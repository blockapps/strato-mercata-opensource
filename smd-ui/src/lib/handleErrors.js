import { env } from "../env";

const HTTP_UNAUTHORIZED = 401;

export function handleErrors(response) {
  // Handled: UNAUTHORIZED and OAUTH_ENABLED
  if (env.OAUTH_ENABLED && response.status === HTTP_UNAUTHORIZED) {
    window.location.reload();
  }
  return response;
}