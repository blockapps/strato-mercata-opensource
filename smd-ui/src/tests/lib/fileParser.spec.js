import { getFileAndReplaceWithImport, getImportStatements, replaceImportStatementsWithSource } from '../../lib/FileParser'

const code = 'import "Greeter.sol"'
const tabData = [
  {
    text: "import \"Greeter.sol\"", title: "Main.sol"
  },
  {
    text: "contract GreeterW {↵    /* Define variable greeting of the type string */↵    string greeting;↵↵    /* This runs when the contract is executed */↵    function GreeterW(string _greeting) public {↵        greeting = _greeting;↵    }↵↵    /* Main function */↵    function greet(string _greeting) constant returns (string) {↵        return greeting;↵    }↵}↵",
    title: "Greeter.sol"
  }
]
const source = "import \"'main.sol'\";\n↵↵contract Cloner {↵Sheep[] public clones;↵string public sheepDNA = \"ccc\";↵↵function cloneSheep(string name) returns (address)↵{ Sheep sheep = new Sheep(name, sheepDNA); clones.push(sheep); return sheep }↵↵}↵↵contract Sheep {↵string public name;↵string public dna;↵↵function Sheep(string _name, string _dna)↵{ name = _name; dna = _dna; }↵↵function geneticallyModify(string _dna)↵{ dna = _dna; }↵↵}"


describe('Lib: file parser', () => {

  test('import statements', () => {
    expect(getImportStatements(source)).toMatchSnapshot();
  });

  test('import code', () => {
    expect(getFileAndReplaceWithImport(code, tabData)).toMatchSnapshot();
  });

});