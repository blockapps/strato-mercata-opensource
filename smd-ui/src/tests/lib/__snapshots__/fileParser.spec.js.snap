// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lib: file parser import code 1`] = `"contract GreeterW {↵    /* Define variable greeting of the type string */↵    string greeting;↵↵    /* This runs when the contract is executed */↵    function GreeterW(string _greeting) public {↵        greeting = _greeting;↵    }↵↵    /* Main function */↵    function greet(string _greeting) constant returns (string) {↵        return greeting;↵    }↵}↵"`;

exports[`Lib: file parser import statements 1`] = `
Array [
  "main.sol",
]
`;
